### DISCUSSION TITLE
Container Object Storage Interface (COSI) Support

### DISCUSSION THREAD

**Discussion by pschichtel** [minio/minio#20759] [Q&A] [answered]
I've been looking to utilize COSI to automate provisioning of object buckets for applications as part of a more dynamic landscape.

I've found the [retired minio driver](https://blog.min.io/kubernetes-csi-and-cosi-a-symbiotic-relationship/) that was apparently built as a PoC and I found [the blog](https://blog.min.io/kubernetes-csi-and-cosi-a-symbiotic-relationship/) post from you guys talking about the relation ship between CSI and COSI, that introduces DirectCSI but doesn't seem to mention anything about a concrete COSI driver for minio.

So the direct question: Does MinIO provide a COSI driver? And if so, can someone refer me to the relevant documentation?

**Comment by harshavardhana** [comment, marked as answer]
@pschichtel COSI was our initiative in k8s but beyond that there was no real traction in the k8s community. We have instead gravitated towards a more homegrown aistor offering that provides this fully self contained.

We are not planning to do this as a community release.

**Reply by pschichtel** [reply]
interesting. there are a couple of drivers available (e.g. for ceph rgw) and I know other commercial storage vendors that at least claim to have support for it. So from from minio's side there is no declarative way of requesting buckets available beyond AIStor? I have little to no use-case for the AIStor offerings in the projects I work on.

**Comment by pschichtel** [comment]
I just found https://github.com/benfiola/minio-operator-ext via https://github.com/minio/operator/issues/1100 which seems to be what I was looking for. It's not COSI, but at the end of the day I don't really care if it is for my use-case.
